src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -33,7 +33,7 @@ public void Initialize(EntityInfo entity, Func<IncomingMessageDetails, ReceiveCo
             receiveMode = settings.Get<ReceiveMode>(WellKnownConfigurationKeys.Connectivity.MessageReceivers.ReceiveMode);
 
             incomingCallback = callback;
-            this.errorCallback = errorCallback;
+            this.errorCallback = errorCallback ?? EmptyErrorCallback;
```

Comment:
The `OnError` method is a public separate method that can be called or not. This propagates down to this point. We should be defensive here. I agree with @danielmarbach